confdb/confdb.go

```diff
@@ -465,47 +533,150 @@ type validationOptions struct {
 }
 
 // validateViewDottedPath validates that request/storage strings in a view definition are:
-//   - composed of non-empty, dot-separated subkeys with optional placeholders ("foo.{bar}"),
-//     if allowed by the validationOptions
+//   - composed of non-empty, dot or bracket separated subkeys with optional
+//     placeholders (e.g., foo.{bar}, a[{n}].bar), if allowed by the validationOptions
 //   - non-placeholder subkeys are made up of lowercase alphanumeric ASCII characters,
 //     optionally with dashes between alphanumeric characters (e.g., "a-b-c")
 //   - placeholder subkeys are composed of non-placeholder subkeys wrapped in curly brackets
-func validateViewDottedPath(path string, opts *validationOptions) (err error) {
+//   - bracketed subkeys that aren't placeholders can only contain integers
+//
+// If the validation succeeds, it returns an []accessor which contains typed
+// representations of each type of subkey (e.g., key placeholder, index, etc).
+func validateViewDottedPath(path string, opts *validationOptions) ([]accessor, error) {
```

Comment:
I'm slightly confused by how the work is distributed between this, splitViewPath and pathIntoAccessors ? as it seems the loop here could do what pathIntoAccessors is doing?

Reply:
Initially I thought it might be necessary to have pathIntoAccessors as a standalone to use in other places. I've merged the accessor creation into this now